rust/noosphere-api/src/data.rs

```diff
@@ -17,71 +17,112 @@ use ucan::{
-    pub actual_tip: Option<Cid>,
+/// The possible responses from the "fetch" API route
+#[derive(Debug, Serialize, Deserialize)]
+pub enum FetchResponse {
+    /// There are new revisions to the local and "counterpart" spheres to sync
+    /// with local history
+    NewChanges {
+        /// The tip of the "counterpart" sphere that is managed by the API host
+        /// that the client is fetching from
+        tip: Cid,
+        /// All the new blocks of the "counterpart" sphere as well as the new
+        /// blocks of the local sphere that correspond to remote changes from
+        /// other clients
+        blocks: Bundle,
+    },
+    /// There are no new revisions since the revision specified in the initial
+    /// fetch request
+    UpToDate,
 }
 
-// Push
+/// The body payload expected by the "push" API route
 #[derive(Debug, Serialize, Deserialize)]
 pub struct PushBody {
+    /// The DID of the local sphere whose revisions are being pushed
     pub sphere: String,
+    /// The base revision represented by the payload being pushed; if the
+    /// entire history is being pushed, then this should be None
     pub base: Option<Cid>,
+    /// The tip of the history represented by the payload being pushed
     pub tip: Cid,
+    /// A bundle of all the blocks needed to hydrate the revisions from the
+    /// base to the tip of history as represented by this payload
     pub blocks: Bundle,
 }
 
+/// The possible responses from the "push" API route
 #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub enum PushResponse {
-    Ok,
-    OutOfDate(OutOfDateResponse),
-    MissingRevisions(MissingRevisionsResponse),
+    /// The new history was accepted
+    Accepted {
+        /// This is the new tip of the "counterpart" sphere after accepting
+        /// the latest history from the local sphere. This is guaranteed to be
+        /// at least one revision ahead of the latest revision being tracked
+        /// by the client (because it points to the newly received tip of the
+        /// local sphere's history)
+        new_tip: Cid,
+        /// The blocks needed to hydrate the revisions of the "counterpart"
+        /// sphere history to the tip represented in this response
+        blocks: Bundle,
+    },
+    /// The history was already known by the API host, so no changes were made
+    NoChange,
 }
 
+/// The response from the "identify" API route; this is a signed response that
+/// allows the client to verify the authority of the API host
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct IdentifyResponse {
+    /// The DID of the API host
     pub gateway_identity: String,
+    /// The DID of the "counterpart" sphere
     pub sphere_identity: String,
+    /// The signature of the API host over this payload, as base64-encoded bytes
     pub signature: String,
+    /// The proof that the API host was authorized by the "counterpart" sphere
+    /// in the form of a UCAN JWT
     pub proof: String,
```

Comment:
This is just an opinion, but I tend to avoid aliased types for this kind of case. It sucks to be stringly typed, but if a type is a string I want the user to know it's a string. A type alias doesn't actually change the underlying type, and may be confusing to read ("ohhh, FoobarKind is actually just a string").

One alternative I like in Rust is to use `#[repr(transparent)]` newtypes, because at least this has type system implications that will force a writer to consciously wrap the string in a thing that has a name. But, this has the downside of requiring some kind of unwrapping or dereferencing to get the underlying type.

Reply:
"stringly typed" :laughing: 

I'm not sure of the extent of these representations to weigh the options, but great points! Wrappers seem non-ideal (TIL `#[repr(transparent)]`). VSCode at least shows the alias upon inspection, but ya some mental overhead there

Reply:
I'm going to move the change forward, but let's track this for future discussion, because we may just be happier overall if we prefer type aliases in the long-run.

I filed this issue for further discussion: https://github.com/subconsciousnetwork/noosphere/issues/108